src/waveWorker.js

```diff
@@ -56,42 +56,35 @@ WavePCM.prototype.record = function( buffers ){
     for ( var channel = 0; channel < this.numberOfChannels; channel++ ) {
 
       var outputIndex = ( i * this.numberOfChannels + channel ) * this.bytesPerSample;
-      var sample = buffers[ channel ][ i ];
 
-      // Check for clipping
-      if ( sample > 1 ) {
-        sample = 1;
-      }
-
-      else if ( sample < -1 ) {
-        sample = -1;
-      }
+      // clip the signal if it exceeds [-1, 1]
+      var sample = Math.max(-1, Math.min(1, buffers[ channel ][ i ]));
 
-      // bit reduce and convert to signed integer
+      // bit reduce and convert to integer
       switch ( this.bytesPerSample ) {
-        case 4: // 32 bits
-          sample = sample * 2147483648;
+        case 4: // 32 bits signed
+          sample = sample * (sample < 0 ? 2147483648 : 2147483647);
```

Comment:
This approach is a little odd since the range distribution on either size of 0 will be unequal.  Of course for 32 bit the difference is negligible, but for 8 bit it seems significant.

Other approaches are to clamp to max of appropriate value less than 1.0 as I mentioned in the bug, or scale both negative and positive to only 2^(N-1) as is done in OpenGL (https://www.khronos.org/opengl/wiki/Normalized_Integer#Signed).

Reply:
Clamping to `1-Epsilon` is clipping valid value of 1 unnecessarily. Doing it this way we take the entire valid sample range of [-1,1] and transform it to the entire valid int range. I am assuming that the wave format playback has accounted for the asymmetrical range on either side of 0 for signed integers, however I couldn't find any spec about this.  

I would be curious to see how well known converters handle this.  Perhaps sox.

Reply:
The sox man page option for `-bits` implies that they are using the entire range of -32768  to 32767. http://sox.sourceforge.net/sox.html

Reply:
Small post covering the different options for conversion:  http://patrakov.blogspot.jp/2010/05/converting-audio-samples.html

It mentions that JACK follows convention I cited for OpenGL, while ALSA follows the clamping to 1-Epsilon approach.

In the comments a 4th approach is mentioned:   `round(sample * 32767.5 - 0.5)`

It's probably worth confirming how Uint8Array handles rounding / truncation from floating point to integer.

Reply:
@belm0 8-bit wave files use unsigned ints. I'm started reading in DSP forums about rounding errors converting floats to ints. https://www.dsprelated.com/showthread/comp.dsp/71057-1.php . :) Perhaps it's time for bed.

Reply:
I confirmed that the JS typed arrays follow C convention of truncating towards 0:
```
$ node
> a = new Int16Array(1)
Int16Array [ 0 ]
> a[0] = 0.7; a[0]
0
> a[0] = -0.7; a[0]
0
```
Given this, you could follow the last conversion given in  https://www.cs.cmu.edu/~rbd/papers/cmj-float-to-int.html.
```sample = sample * 32767.5 - 0.5```
So i.e. -1.0 yields -32768, 1.0 yields 32767, and 0.0 yields -0.5 which truncates to 0.